src/aci/aci.h

```diff
@@ -332,15 +332,33 @@ class AdaptiveCI : public Wavefunction {
                                    det_hash<std::vector<double>>& V_hash);
     /// Get excited determinants with a specified hole
     void get_core_excited_determinants(SharedMatrix evecs, DeterminantHashVec& P_space,
-                                   det_hash<std::vector<double>>& V_hash);
+                                       det_hash<std::vector<double>>& V_hash);
     void get_excited_determinants_sr(SharedMatrix evecs, DeterminantHashVec& P_space,
-                                   det_hash<double>& V_hash);
-    double get_excited_determinants_batch(SharedMatrix evecs, SharedVector evals, DeterminantHashVec& P_space,
-                                   std::vector<std::pair<double, Determinant>>& F_space);
+                                     det_hash<double>& V_hash);
+    double get_excited_determinants_batch(SharedMatrix evecs, SharedVector evals,
+                                          DeterminantHashVec& P_space,
+                                          std::vector<std::pair<double, Determinant>>& F_space);
+    double get_excited_determinants_batch2(SharedMatrix evecs, SharedVector evals,
+                                           DeterminantHashVec& P_space,
+                                           std::vector<std::pair<double, Determinant>>& F_space);
+    double get_excited_determinants_batch3(SharedMatrix evecs, SharedVector evals,
+                                           DeterminantHashVec& P_space,
+                                           std::vector<std::pair<double, Determinant>>& F_space);
+    double get_excited_determinants_batchv(SharedMatrix evecs, SharedVector evals,
+                                           DeterminantHashVec& P_space,
+                                           std::vector<std::pair<double, Determinant>>& F_space);
 
     /// Builds excited determinants in batch
     det_hash<double> get_bin_F_space(int bin, int nbin, SharedMatrix evecs,
-                                    DeterminantHashVec& P_space); 
+                                     DeterminantHashVec& P_space);
+    det_hash<double> get_bin_F_space2(int bin, int nbin,double E0, SharedMatrix evecs,
+                                      DeterminantHashVec& P_space);
+    std::pair<std::vector<std::vector<std::pair<Determinant, double>>>, std::vector<size_t>>
+    get_bin_F_space3(int bin, int nbin, SharedMatrix evecs, DeterminantHashVec& P_space);
+    std::vector<std::pair<Determinant, double>>
+    get_bin_F_spacev(int bin, int nbin, SharedMatrix evecs, DeterminantHashVec& P_space);
```

Comment:
What do these '1','2','3','v' mean after `get_excited_determinants_batch` and `get_bin_F_space`?
Can be more readable if you use more self-explanatory function names or add comments.

Reply:
Good point. @jeffschriber, could you fix this? I am responsible for `get_excited_determinants_batch3` I think. Can you change this to `get_excited_determinants_batch_sort`?